base/datatype_reference.bal

```diff
@@ -78,3 +78,71 @@ public type Reference record {|
     Identifier identifier?;
     string display?;
 |};
+
```

Comment:
Isn't the logic common among multiple functions? Can't it be extracted out to a function instead?

Reply:
Code removed due to non-relevant code in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/817dadb5178bde7d809333a26f465c6688d3879a